velox/exec/MergeJoin.cpp

```diff
@@ -273,11 +290,18 @@ void MergeJoin::prepareOutput() {
     outputSize_ = 0;
 
     if (filterInput_ != nullptr) {
-      // When filterInput_ contains array or map columns, their child vectors
-      // (elements, keys and values) keep growing after each call to
-      // 'copyRow'. Call BaseVector::resize(0) on these child vectors to avoid
-      // that.
-      for (auto& child : filterInput_->children()) {
+      for (auto i = 0; i < filterInputType_->size(); ++i) {
+        auto& child = filterInput_->childAt(i);
```

Comment:
It took me awhile to understand what this logic is doing.

I think it makes sure that filterInput_ and output_ use the same shared pointers for the columns which are used in the filter and projected out.

Reply:
That's true, I've updated the comment to reflect that.